jxl/src/var_dct/dct.rs

```diff
@@ -313,48 +284,20 @@ where
 {
     assert_eq!(data.len(), ROWS * COLS, "Data length mismatch");
 
-    // Create a temporary flat buffer for the transposed data.
-    let mut transposed_data = vec![0.0f32; ROWS * COLS];
+    // Create a temporary buffer for the transposed data.
```

Comment:
we should probably pass a temporary buffer in as we do in libjxl, btw

Reply:
Ah, you mean a preallocated “scratch” buffer like we have in various places?